### DISCUSSION TITLE
How to contribute a tree-sitter grammar for a language so that GH highlights it?

### DISCUSSION THREAD

**Discussion by jjant** [github-linguist/linguist#6073] [Q&A] [answered]
I would like to have syntax highlighting for [`Smithy`](https://awslabs.github.io/smithy/2.0/index.html) files in GH.

It looks like GH supports using tree-sitter grammars for highlighting, but I couldn't find if they accept new language contributions, or if they do, where to make them. Can anyone clarify?

**Comment by lildude** [comment, marked as answer]
There isn't currently a public process adding a tree-sitter grammar as this isn't handled by Linguist. This is handled internally by another team that controls the syntax highlighting service. The best I can suggest is starting a discussion at https://github.community or opening a support ticket using the "Contact GitHub" link at the bottom of any page on GitHub asking for this to be considered.

If you have a TextMate compatible grammar, we can certainly add support for the language (we'll need to do this anyway) and use that via Linguist in the mean time, assuming the new language meets the documented popularity requirements. See the CONTRIBUTING.md file for more deets. If and when a Treesitter grammar is implemented, it'll automatically take over from the TextMate grammar.

**Reply by Baccata** [reply]
A tree-sitter grammar for Smithy (compatible with the 2.0 syntax) exists there : https://github.com/indoorvivants/tree-sitter-smithy. 

@lildude I presume linguist supports pointing to third-party tree-sitter grammars (as long as the relevant repo is correctly added as a submodule and accompanied with a bit of configuration). Can you confirm ? 

I've been looking at some PRs to understand what's required, I'd be happy to raise the PR myself to point to that third-party grammar.

**Reply by lildude** [reply]
You presume incorrectly 😁 Linguist only points to Textmate-compatible grammars. All tree-sitter grammars are handled separate directly in the internal syntax highlighting engine.

If you see a reference to a tree-sitter grammar in the vendor/README.md file it’s just an indication that Linguist’s linked grammar isn’t being used on GitHub for that language, even though the Textmate grammar is still shipped with Linguist.

**Reply by Baccata** [reply]
haha fair enough 😄  @jjant are you happy to take this on (ie PR linguist, pointing to the textmate grammar) ? I was genuinely gonna tackle it this week but you seem to be part of the `awslabs` org, so I'd happily defer to you if you prefer. I'm just looking for  something better than the current state, whilst a tree-sitter based solution is proposed/reviewed/actioned.

**Reply by jjant** [reply]
I've created a PR to support Smithy here: https://github.com/github/linguist/pull/6110.
CC @Baccata @lildude.
